core/commands/add.go

```diff
@@ -227,9 +250,24 @@ func add(n *core.IpfsNode, reader io.Reader) (*dag.Node, error) {
 	return node, nil
 }
 
-func addFile(n *core.IpfsNode, file files.File, out chan interface{}, progress bool, wrap bool) (*dag.Node, error) {
+func addFile(n *core.IpfsNode, file files.File, out chan interface{}, progress bool, wrap bool, hidden bool, ignoreFilePatterns []ignore.GitIgnore) (*dag.Node, error) {
```

Comment:
this function is getting crazy. it may be useful to construct an `adder` object that holds the configuration:

```
type adder struct {
  n *core.IpfsNode
  progress bool
  wrap bool
  hidden bool
  ignoreFilePatterns []ignore.GitIgnore
}

func (adr *adder) addFile(file files.File) (*dag.Node, error) { ... }
```